server/events/vcs/gitlab_client.go

```diff
@@ -405,10 +405,31 @@ func gitlabIsMergeable(mr *gitlab.MergeRequest, project *gitlab.Project, support
 	}
 
 	if supportsDetailedMergeStatus {
+		if mr.DetailedMergeStatus == "need_rebase" {
+			// https://docs.gitlab.com/api/projects/#project-merge-method
+			// +----------------------+---------------+---------------------------------------------+
+			// | Constant Name        | API value     | Description in UI                           |
+			// +----------------------+---------------+---------------------------------------------+
+			// | NoFastForwardMerge   | merge         | Merge Commit                                |
+			// | RebaseMerge          | rebase_merge  | Merge commit with semi-linear history       |
+			// | FastForwardMerge     | ff            | Fast-forward merge                          |
+			// +----------------------+---------------+---------------------------------------------+
+			//
+			// If an MR needs to be rebased (i.e., is behind), and the project is in RebaseMerge or FastForwardMerge state
+			// gitlab doesn't allow you to merge it, and we shouldn't either
+			if project.MergeMethod == gitlab.RebaseMerge || project.MergeMethod == gitlab.FastForwardMerge {
+				return models.MergeableStatus{
+					IsMergeable: false,
+					Reason:      fmt.Sprintf("Merge status is need_rebase and merge method is %s", project.MergeMethod),
+				}
+			}
+			return models.MergeableStatus{
+				IsMergeable: true,
+			}
+		}
```

Comment:
I think this logic is not needed.
`"detailed_merge_status": "need_rebase"` is returned only if the merge method is Fast-forward or Rebase.

If the merge method is Merge commit, the API returns` "detailed_merge_status": "mergeable",` even if a rebase is possible.

Reply:
Ah I see! I was able to confirm that, I'll fix up the code.